interactions/api/http.py

```diff
@@ -0,0 +1,2285 @@
+from asyncio import AbstractEventLoop, Event, Lock, get_event_loop, sleep
+from logging import Logger, basicConfig, getLogger
+from sys import version_info
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, Union
+from urllib.parse import quote
+
+from aiohttp import ClientSession, FormData
+from aiohttp import __version__ as http_version
+
+from ..api.error import HTTPException
+from ..api.models import (
+    Channel,
+    Embed,
+    Emoji,
+    Guild,
+    GuildPreview,
+    GuildTemplate,
+    Invite,
+    Member,
+    Message,
+    Role,
+    StageInstance,
+    User,
+    VoiceRegion,
+    WelcomeScreen,
+)
+from ..base import Data, __version__
+
+basicConfig(level=Data.LOGGER)
+log: Logger = getLogger("http")
+
+__all__ = ("Route", "Padlock", "Request", "HTTPClient")
+session: ClientSession = ClientSession()
+
+
+class Route:
+    """
+    A class representing how an HTTP route is structured.
+
+    :ivar ClassVar[str] __api__: The HTTP route path.
+    :ivar str method: The HTTP method.
+    :ivar str path: The URL path.
+    :ivar Optional[str] channel_id: The channel ID from the bucket if given.
+    :ivar Optional[str] guild_id: The guild ID from the bucket if given.
+    """
+
+    __slots__ = ("__api__", "method", "path", "channel_id", "guild_id")
+    __api__: ClassVar[str]
+    method: str
+    path: str
+    channel_id: Optional[str]
+    guild_id: Optional[str]
+
+    def __init__(self, method: str, path: str, **kwargs) -> None:
+        r"""
+        :param method: The HTTP request method.
+        :type method: str
+        :param path: The path of the HTTP/URL.
+        :type path: str
+        :param \**kwargs: Optional keyword-only arguments to pass as information in the route.
+        :type \**kwargs: dict
+        """
+        self.__api__ = "https://discord.com/api/v9"
+        self.method = method
+        self.path = path.format(**kwargs)
+        self.channel_id = kwargs.get("channel_id")
+        self.guild_id = kwargs.get("guild_id")
+
+    @property
+    def bucket(self) -> str:
+        """
+        Returns the route's bucket.
+
+        :return: The route bucket.
+        :rtype: str
+        """
+        return f"{self.channel_id}:{self.guild_id}:{self.path}"
```

Comment:
`webhook_id + webhook_token` should also be used for bucketing interaction requests (iirc blue requested this a while back)